lib/grp.gi

```diff
@@ -4398,55 +4398,48 @@ end);
 InstallMethod( GroupWithGenerators,
     "generic method for collection",
     [ IsCollection ],
-function( gens )
-local G,typ;
+    function( gens )
+    local typ;
 
-  typ:=MakeGroupyType(FamilyObj(gens),
+    gens:= AsList( gens );
+    typ:= MakeGroupyType( FamilyObj( gens ),
           IsGroup and IsAttributeStoringRep 
           and HasIsEmpty and HasGeneratorsOfMagmaWithInverses,
           gens,false,true);
 
-  G:=rec();
-  ObjectifyWithAttributes(G,typ,GeneratorsOfMagmaWithInverses,AsList(gens));
-
-  return G;
+    return ObjectifyWithAttributes( rec(), typ,
+               GeneratorsOfMagmaWithInverses, gens );
 end );
```

Comment:
This completely changes formatting, but (as far as I could tell) really only inserts a single, line, `gens:= AsList( gens );`, and then replaces one `AsList(gens)` later by `gens`.

I'd really prefer if this changes was kept to the minimum, for ease of reviewing.

Reply:
Note that `gens` is used by `MakeGroupyType`.
This is where the error happens in the old version.